ratis-metrics/src/main/java/org/apache/ratis/metrics/impl/RatisMetricRegistryImpl.java

```diff
@@ -116,4 +123,34 @@ private String getMetricName(String shortName) {
       }
     }
   }
+
+  @Override
+  public void setJmxReporter(JmxReporter jmxReporter) {
+    this.jmxReporter = jmxReporter;
+  }
+
+  @Override
+  public JmxReporter getJmxReporter() {
+    return this.jmxReporter;
+  }
+
+  @Override
+  public void setConsoleReporter(ConsoleReporter consoleReporter) {
+    this.consoleReporter = consoleReporter;
+  }
+
+  @Override
+  public ConsoleReporter getConsoleReporter() {
+    return this.consoleReporter;
+  }
+
+  @Override
+  public void setPrometheusCollector(Collector prometheusCollector) {
```

Comment:
Are we doing this to support a prometheus collector purely in Ratis, instead of depending on an external client like Ozone?

Reply:
@avijayanhwx Thanks for review. Actually, I also do not want to add prometheus in Ratis.
But because ozone pass a lambda to ratis to register collector in prometheus, and if we want to unregister the collector when remove registry by `CollectorRegistry.defaultRegistry.unregister(collector)`, we must record the collector in Ratis, if you have better idea, plz share it.  The following code copy from ozone related PR: https://github.com/apache/hadoop-ozone/pull/885.
![image](https://user-images.githubusercontent.com/51938049/80743904-62d51e00-8b50-11ea-93bd-91630c1f5877.png)

Reply:
@avijayanhwx Thanks for review. I have remove prometheus in Ratis, I handle this in ozone.